test/basic_json.cpp

```diff
@@ -0,0 +1,1480 @@
+    EXPECT_EQ(before.key(), "key");
+    EXPECT_EQ(before.val(), "val");
+    root["b"]["seq2"][1] = "01";
+    NodeRef after = root["b"]["key"];
+    EXPECT_EQ(before.key(), "key");
+    EXPECT_EQ(before.val(), "val");
+    EXPECT_EQ(after.key(), "key");
+    EXPECT_EQ(after.val(), "val");
+    root["b"]["seq2"][2] = "02";
+    root["b"]["seq2"][3] = "03";
+    int iv = 0;
+    root["b"]["seq2"][4] << 55; root["b"]["seq2"][4] >> iv;
+    size_t zv = 0;
+    root["b"]["seq2"][5] << size_t(55); root["b"]["seq2"][5] >> zv;
+    float fv = 0;
+    root["b"]["seq2"][6] << 2.0f; root["b"]["seq2"][6] >> fv;
+    float dv = 0;
+    root["b"]["seq2"][7] << 2.0; root["b"]["seq2"][7] >> dv;
+
+    EXPECT_EQ(root["b"]["key"].key(), "key");
+    EXPECT_EQ(root["b"]["key"].val(), "val");
+
+
+    emit_json(t);
+
+    EXPECT_EQ(root.type(), MAP);
+    EXPECT_EQ(root["a"].type(), KEYVAL);
+    EXPECT_EQ(root["a"].key(), "a");
+    EXPECT_EQ(root["a"].val(), "0");
+
+    EXPECT_EQ(root["b"].type(), KEYMAP);
+
+    EXPECT_EQ(root["b"]["seq"].type(), KEYSEQ);
+    EXPECT_EQ(root["b"]["seq"].key(), "seq");
+    EXPECT_EQ(root["b"]["seq"][0].type(), VAL);
+    EXPECT_EQ(root["b"]["seq"][0].val(), "0");
+    EXPECT_EQ(root["b"]["seq"][1].type(), VAL);
+    EXPECT_EQ(root["b"]["seq"][1].val(), "1");
+    EXPECT_EQ(root["b"]["seq"][2].type(), VAL);
+    EXPECT_EQ(root["b"]["seq"][2].val(), "2");
+    EXPECT_EQ(root["b"]["seq"][3].type(), VAL);
+    EXPECT_EQ(root["b"]["seq"][3].val(), "3");
+    EXPECT_EQ(root["b"]["seq"][3].val_tag(), "!!str");
+    EXPECT_EQ(root["b"]["seq"][4].type(), VAL);
+    EXPECT_EQ(root["b"]["seq"][4].val(), "4");
+
+    int tv;
+    EXPECT_EQ(root["b"]["key"].key(), "key");
+    EXPECT_EQ(root["b"]["key"].val(), "val");
+    EXPECT_EQ(root["b"]["seq2"][0].val(), "00"); root["b"]["seq2"][0] >> tv; EXPECT_EQ(tv, 0);
+    EXPECT_EQ(root["b"]["seq2"][1].val(), "01"); root["b"]["seq2"][1] >> tv; EXPECT_EQ(tv, 1);
+    EXPECT_EQ(root["b"]["seq2"][2].val(), "02"); root["b"]["seq2"][2] >> tv; EXPECT_EQ(tv, 2);
+    EXPECT_EQ(root["b"]["seq2"][3].val(), "03"); root["b"]["seq2"][3] >> tv; EXPECT_EQ(tv, 3);
+    EXPECT_EQ(root["b"]["seq2"][4].val(), "55"); EXPECT_EQ(iv, 55);
+    EXPECT_EQ(root["b"]["seq2"][5].val(), "55"); EXPECT_EQ(zv, size_t(55));
+    EXPECT_EQ(root["b"]["seq2"][6].val(), "2"); EXPECT_EQ(fv, 2.f);
+    EXPECT_EQ(root["b"]["seq2"][6].val(), "2"); EXPECT_EQ(dv, 2.);
+
+    root["b"]["seq"][2].set_val_serialized(22);
+
+    emit_json(t);
+
+    EXPECT_EQ(root["b"]["aaa"].type(), KEYVAL);
+    EXPECT_EQ(root["b"]["aaa"].key_tag(), "!!str");
+    EXPECT_EQ(root["b"]["aaa"].key(), "aaa");
+    EXPECT_EQ(root["b"]["aaa"].val_tag(), "!!int");
+    EXPECT_EQ(root["b"]["aaa"].val(), "0");
+}
+
+//-----------------------------------------------------------------------------
+//-----------------------------------------------------------------------------
+//-----------------------------------------------------------------------------
```

Comment:
from here on, apparently all test are simple duplicates of the original, and test nothing specific to json emission.

Please remove.

Reply:
done